optimum/neuron/distributed/parallel_layers.py

```diff
@@ -705,3 +707,202 @@ def transform(
         cls.patch_cross_entropy(model)
 
         return layer
+
+
+@dataclass(frozen=True)
+class SequenceCollectiveOpInfo:
+    """
+    Represents a collective op to perform.
+
+    Example:
+
+        ```python
+        info = SequenceCollectiveOpInfo("gather", torch.nn.LayerNorm, "output", "last")
+        ```
+        Here `info` indicates that the output of the last `torch.nn.LayerNorm` must be gathered on the sequence axis.
+
+    """
+
+    collective_op: Union[Literal["scatter"], Literal["gather"]]
+    layer: Union[Type["torch.nn.Module"], str]
+    io: Union[Literal["input"], Literal["output"]]
+    first_or_last: Union[Literal["first"], Literal["last"]]
```

Comment:
Why not just use a boolean ?

Reply:
Because I find it easier to understand:

```python
SequenceCollectiveOpInfo("scatter", torch.nn.LayerNorm, "input", "first"),
```

Is more explicit than:

```python
SequenceCollectiveOpInfo(True, torch.nn.LayerNorm, False, True),
```